### PR TITLE
V1.4 refactor

### LINKED ISSUE #271: Final code refactoring
The following should be done, preferably after all code has been finalised.

- [x] Remaining "Person"s to "Student"s
- [x] "AddressBook" to "SourceControl"
- [x] Rename package `seedu.address`
   - [x] Ensure that links in `DeveloperGuide` are not broken after change
   - [x] Update diagrams accordingly
- [x] Reorganise packages within `seedu.address.model.student` into their own respective packages

### PR DESCRIPTION
**Pull request by jonas-chow**
Closes #271 

All AddressBook renamed to SourceControl, all Person renamed to Student.
Checked: All java, xml, md files.
Also verified that all commands work as expected.

For easy reference: the student package now looks like this:
Screenshot 2021-11-07 at 3 41 13 AM

### CODE REVIEW

File: src/main/java/seedu/sourcecontrol/storage/JsonSourceControlStorage.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    /**
+     * Similar to {@link #readSourceControl()}.
+     *
+     * @param filePath location of the data. Cannot be null.
+     * @throws DataConversionException if the file is not in the correct format.
+     */
+    public Optional<ReadOnlySourceControl> readSourceControl(Path filePath) throws DataConversionException {
+        requireNonNull(filePath);
+
+        Optional<JsonSerializableSourceControl> jsonAddressBook = JsonUtil.readJsonFile(
```
Comment (ganhongyao) [inline src/main/java/seedu/sourcecontrol/storage/JsonSourceControlStorage.java]:
```suggestion
        Optional<JsonSerializableSourceControl> jsonSourceControl = JsonUtil.readJsonFile(
```

File: src/main/java/seedu/sourcecontrol/storage/JsonSourceControlStorage.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                filePath, JsonSerializableSourceControl.class);
+        if (!jsonAddressBook.isPresent()) {
```
Comment (ganhongyao) [inline src/main/java/seedu/sourcecontrol/storage/JsonSourceControlStorage.java]:
```suggestion
        if (!jsonSourceControl.isPresent()) {
```

File: src/main/java/seedu/sourcecontrol/storage/JsonSourceControlStorage.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            return Optional.empty();
+        }
+
+        try {
+            return Optional.of(jsonAddressBook.get().toModelType());
```
Comment (ganhongyao) [inline src/main/java/seedu/sourcecontrol/storage/JsonSourceControlStorage.java]:
```suggestion
            return Optional.of(jsonSourceControl.get().toModelType());
```

File: src/test/java/seedu/sourcecontrol/model/ModelManagerTest.java [hunk 4]
```diff
@@ -146,8 +146,8 @@ public void getAssessment() {
 
     @Test
     public void equals() {
-        AddressBook addressBook = new AddressBookBuilder().withPerson(ALICE).withPerson(BENSON).build();
-        AddressBook differentAddressBook = new AddressBook();
+        SourceControl addressBook = new SourceControlBuilder().withStudent(ALICE).withStudent(BENSON).build();
```
Comment (ganhongyao) [inline src/test/java/seedu/sourcecontrol/model/ModelManagerTest.java]:
```suggestion
        SourceControl sourceControl = new SourceControlBuilder().withStudent(ALICE).withStudent(BENSON).build();
```
should we change these as well? same for the line below

File: src/test/java/seedu/sourcecontrol/model/ModelManagerTest.java [hunk 5]
```diff
@@ -68,34 +68,34 @@ public void setGuiSettings_validGuiSettings_setsGuiSettings() {
 
     @Test
     public void setAddressBookFilePath_nullPath_throwsNullPointerException() {
```
Comment (ganhongyao) [inline src/test/java/seedu/sourcecontrol/model/ModelManagerTest.java]:
```suggestion
    public void setSourceControlFilePath_nullPath_throwsNullPointerException() {
```
should we change the method names too? same for the other methods in this file

File: src/test/java/seedu/sourcecontrol/logic/commands/EditCommandTest.java [hunk 6]
```diff
@@ -1,121 +1,121 @@
...
         assertCommandFailure(editCommand, model, EditCommand.MESSAGE_DUPLICATE_STUDENT);
     }
 
     @Test
-    public void execute_duplicatePersonFilteredList_failure() {
-        showPersonAtIndex(model, INDEX_FIRST_PERSON);
+    public void execute_duplicateStudentFilteredList_failure() {
+        showStudentAtIndex(model, INDEX_FIRST_STUDENT);
 
         // edit student in filtered list into a duplicate in address book
```
Comment (ganhongyao) [inline src/test/java/seedu/sourcecontrol/logic/commands/EditCommandTest.java]:
```suggestion
        // edit student in filtered list into a duplicate in main student list
```

File: src/test/java/seedu/sourcecontrol/logic/commands/EditCommandTest.java [hunk 7]
```diff
@@ -144,11 +144,11 @@ public void execute_invalidPersonIndexUnfilteredList_failure() {
      * but smaller than size of address book
```
Comment (ganhongyao) [inline src/test/java/seedu/sourcecontrol/logic/commands/EditCommandTest.java]:
```suggestion
     * but smaller than size of main student list
```

File: src/test/java/seedu/sourcecontrol/logic/commands/EditCommandTest.java [hunk 8]
```diff
@@ -144,11 +144,11 @@ public void execute_invalidPersonIndexUnfilteredList_failure() {
...
      */
     @Test
-    public void execute_invalidPersonIndexFilteredList_failure() {
-        showPersonAtIndex(model, INDEX_FIRST_PERSON);
-        Index outOfBoundIndex = INDEX_SECOND_PERSON;
+    public void execute_invalidStudentIndexFilteredList_failure() {
+        showStudentAtIndex(model, INDEX_FIRST_STUDENT);
+        Index outOfBoundIndex = INDEX_SECOND_STUDENT;
         // ensures that outOfBoundIndex is still in bounds of address book list
```
Comment (ganhongyao) [inline src/test/java/seedu/sourcecontrol/logic/commands/EditCommandTest.java]:
```suggestion
        // ensures that outOfBoundIndex is still in bounds of main student list
```

**Review by ganhongyao — CHANGES_REQUESTED** [review]
Took a quick look and all looks good, just a few lines you missed out.

Also, 
- Should the `ID` class name be `Id` instead? Similar to how `Ui` is named
- Is there a reason why `StudentStatistics` is under the `assessment` package but `GroupStatistics` is under the `group` package?

Thanks a lot for helping to refactor 🥇

**Comment by nbvannhi** [comment]
Sr if this is already discussed in yesterday meeting.. I notice that class the `addstudent` command is `AddCommand`, so we're gonna leave it like that?

**Comment by zhing22** [comment]
I believe the aliasCommandParser is also named differently?

File: src/main/java/seedu/sourcecontrol/model/student/Student.java [hunk 9]
```diff
@@ -152,7 +152,7 @@ public int hashCode() {
     public String toString() {
         final StringBuilder builder = new StringBuilder();
         builder.append(getName())
-                .append("; NUSNET ID: ")
+                .append("; NUSNET Id: ")
```
Comment (ganhongyao) [inline src/main/java/seedu/sourcecontrol/model/student/Student.java]:
```suggestion
                .append("; NUSNET ID: ")
```

File: src/main/java/seedu/sourcecontrol/logic/commands/ImportCommand.java [hunk 10]
```diff
@@ -49,7 +49,7 @@
     public static final String MESSAGE_INVALID_NUMBER = "Number of columns should be a non-negative integer. ";
     public static final String MESSAGE_OUT_OF_BOUNDS = "Reached unexpected end of line while reading from file. ";
     public static final String MESSAGE_DUPLICATE_ASSESSMENT = "Duplicate assessment found in file. ";
-    public static final String MESSAGE_DUPLICATE_ID = "Duplicate student ID found in file. ";
+    public static final String MESSAGE_DUPLICATE_ID = "Duplicate student Id found in file. ";
```
Comment (ganhongyao) [inline src/main/java/seedu/sourcecontrol/logic/commands/ImportCommand.java]:
```suggestion
    public static final String MESSAGE_DUPLICATE_ID = "Duplicate student ID found in file. ";
```

File: src/main/java/seedu/sourcecontrol/logic/commands/AddScoreCommand.java [hunk 11]
```diff
@@ -49,7 +49,7 @@
     public static final String MESSAGE_NONEXISTENT_ASSESSMENT = "This assessment does not exist.";
     public static final String MESSAGE_NONEXISTENT_STUDENT = "This student does not exist.";
     public static final String MESSAGE_DUPLICATE_STUDENT_NAME =
-            "Score needs to be added through ID for this student due to duplicate naming.";
+            "Score needs to be added through Id for this student due to duplicate naming.";
```
Comment (ganhongyao) [inline src/main/java/seedu/sourcecontrol/logic/commands/AddScoreCommand.java]:
```suggestion
            "Score needs to be added through ID for this student due to duplicate naming.";
```

File: src/main/java/seedu/sourcecontrol/logic/commands/AddGroupCommand.java [hunk 12]
```diff
@@ -41,10 +41,10 @@
 
     public static final String MESSAGE_SUCCESS = "New group added: %1$s\n";
     public static final String MESSAGE_STUDENTS_ADDED = "Students added to group: %1$s\n";
-    public static final String MESSAGE_NONEXISTENT_STUDENT = "Student with name or ID \"%1$s\" does not exist.";
+    public static final String MESSAGE_NONEXISTENT_STUDENT = "Student with name or Id \"%1$s\" does not exist.";
     public static final String MESSAGE_DUPLICATE_GROUP = "This group already exists in the database.";
     public static final String MESSAGE_DUPLICATE_STUDENT =
-            "The student \"%1$s\" needs to be allocated manually using ID due to duplicate naming.";
+            "The student \"%1$s\" needs to be allocated manually using Id due to duplicate naming.";
```
Comment (ganhongyao) [inline src/main/java/seedu/sourcecontrol/logic/commands/AddGroupCommand.java]:
```suggestion
            "The student \"%1$s\" needs to be allocated manually using ID due to duplicate naming.";
```

File: src/main/java/seedu/sourcecontrol/logic/commands/AddGroupCommand.java [hunk 13]
```diff
@@ -41,10 +41,10 @@
 
     public static final String MESSAGE_SUCCESS = "New group added: %1$s\n";
     public static final String MESSAGE_STUDENTS_ADDED = "Students added to group: %1$s\n";
-    public static final String MESSAGE_NONEXISTENT_STUDENT = "Student with name or ID \"%1$s\" does not exist.";
+    public static final String MESSAGE_NONEXISTENT_STUDENT = "Student with name or Id \"%1$s\" does not exist.";
```
Comment (ganhongyao) [inline src/main/java/seedu/sourcecontrol/logic/commands/AddGroupCommand.java]:
```suggestion
    public static final String MESSAGE_NONEXISTENT_STUDENT = "Student with name or ID \"%1$s\" does not exist.";
```

File: src/main/java/seedu/sourcecontrol/logic/commands/AddCommand.java [hunk 14]
```diff
@@ -31,7 +31,7 @@
             + PREFIX_TAG + "beginner";
 
     public static final String MESSAGE_SUCCESS = "New student added: %1$s";
-    public static final String MESSAGE_DUPLICATE_STUDENT = "This student ID already exists in Source Control. ";
+    public static final String MESSAGE_DUPLICATE_STUDENT = "This student Id already exists in Source Control. ";
```
Comment (ganhongyao) [inline src/main/java/seedu/sourcecontrol/logic/commands/AddCommand.java]:
```suggestion
    public static final String MESSAGE_DUPLICATE_STUDENT = "This student ID already exists in Source Control. ";
```

File: src/main/java/seedu/sourcecontrol/logic/commands/AddAllocCommand.java [hunk 15]
```diff
@@ -45,7 +45,7 @@
     public static final String MESSAGE_NONEXISTENT_GROUP = "This group does not exist.";
     public static final String MESSAGE_NONEXISTENT_STUDENT = "This student does not exist.";
     public static final String MESSAGE_DUPLICATE_STUDENT_NAME =
-            "This student needs to be allocated using ID due to duplicate naming.";
+            "This student needs to be allocated using Id due to duplicate naming.";
```
Comment (ganhongyao) [inline src/main/java/seedu/sourcecontrol/logic/commands/AddAllocCommand.java]:
```suggestion
            "This student needs to be allocated using ID due to duplicate naming.";
```

File: src/main/java/seedu/sourcecontrol/ui/InfoDisplay.java [hunk 16]
```diff
@@ -71,7 +71,7 @@ public void setInfo(Info info) {
      */
     public void setStudentInfo(Student student) {
         name.setText(student.getName().fullName);
-        info1.setText("ID: " + student.getId().value);
+        info1.setText("Id: " + student.getId().value);
```
Comment (ganhongyao) [inline src/main/java/seedu/sourcecontrol/ui/InfoDisplay.java]:
```suggestion
        info1.setText("ID: " + student.getId().value);
```

File: src/main/java/seedu/sourcecontrol/ui/StudentCard.java [hunk 17]
```diff
@@ -52,7 +52,7 @@ public StudentCard(Student student, int displayedIndex) {
         this.student = student;
         id.setText(displayedIndex + ". ");
         name.setText(student.getName().fullName);
-        nusNetId.setText("ID: " + student.getId().value);
+        nusNetId.setText("Id: " + student.getId().value);
```
Comment (ganhongyao) [inline src/main/java/seedu/sourcecontrol/ui/StudentCard.java]:
```suggestion
        nusNetId.setText("ID: " + student.getId().value);
```

File: src/main/java/seedu/sourcecontrol/logic/parser/SearchCommandParser.java [hunk 18]
```diff
@@ -49,7 +49,7 @@ public SearchCommand parse(String args) throws ParseException {
         }
 
         if (!argMultimap.getValue(PREFIX_ID).isEmpty()) {
-            searchType = "ID";
+            searchType = "Id";
```
Comment (ganhongyao) [inline src/main/java/seedu/sourcecontrol/logic/parser/SearchCommandParser.java]: I think this should be fine since the `case` still matches

**Review by ganhongyao — CHANGES_REQUESTED** [review]
Sorry, I think you accidentally replaced the `ID` with `Id` in comments and the strings as well. I think the comments are fine, but the messages should still be `ID`. I highlighted those that needs to be reverted, hopefully I didn't miss out any.

**Comment by ganhongyao** [comment]
I think to be safe you can rollback the changes and re-do the "Refactor", but this time excluding the usages in strings and comments

image

**Comment by ganhongyao** [comment]
> Sr if this is already discussed in yesterday meeting.. I notice that class the `addstudent` command is `AddCommand`, so we're gonna leave it like that?

@jonas-chow I think this will be good too



> I believe the aliasCommandParser is also named differently?

@zhing22 so do you mean we change `AliasCommand` to `AddAliasCommand` and `AliasCommandParser` to `AddAliasCommandParser`?

**Comment by nbvannhi** [comment]
> @zhing22 so do you mean we change `AliasCommand` to `AddAliasCommand` and `AliasCommandParser` to `AddAliasCommandParser`?

Hm I think `AliasCommand` is okay tho since the command keyword is also `alias`

**Comment by jonas-chow** [comment]
Thanks for the feedback guys, made the relevant changes. @ganhongyao hopefully just one last review will do

**Review by ganhongyao — APPROVED** [review]
LGTM! 👑

**Pull request merged**